### ISSUE TITLE
cy.server() and cy.route() in before() wiped out for the next text

### ISSUE THREAD

**Issue by Postavshik** [cypress-io/cypress#4435]
### Current behavior: 

I have found this issues was before and it was resolved according to the issue details, but I still have this
https://github.com/cypress-io/cypress/issues/595

I have two tests and one `before` hook where is set up my routes and making login. First test will pass but second will fail because it can't find routes. In order to make this thing working, I need to put `cy.server()` and `cy.route()` in `beforeEach()` hook, to set my routes for every tests. I feel this is something wrong.

### Desired behavior:

I have to set up `cy.server()` and `cy.route()` just once in the `before()` hook and use these routes for every test in the `describe()`

### Versions
Cypress 3.3.1
OS: macOS Mojave 10.14.5
Electron 61

**Comment by jennifer-shehane** [comment]
I can see how the #595 issue is a bit confusing. I've updated the last comment in that issue to clarify more on what was a bug versus what was intended behavior from that issue. 

The way you are seeing `cy.server()` and `cy.route()` today is intended behavior. Code in a `before` hook only runs once and once only. When it reaches the 2nd test, the server and route have been reset and need to be reinvoked. Moving this to a `beforeEach` is the correct. 

If you want to set this up before every single spec file - look into moving the `beforeEach` hook into your [supportFile](https://on.cypress.io/writing-and-organizing-tests#Support-file).

**Comment by Postavshik** [comment]
@jennifer-shehane 
Looks like it didn't resolve my issue :(
I have random issue when sometimes routes do not work and it crash not just one test, because it happens in beforeEach it skips all the rest tests. I have about 100 tests in one file separated by 7 describes. So out of those 100 tests I have randomly 3-4 tests failed because of this problems. Please check the screenshot.
![2019-06-12_0857](https://user-images.githubusercontent.com/21285622/59359481-08429180-8cf4-11e9-9b8c-6f0f2abfc4b3.png)

**Comment by Postavshik** [comment]
I think I have found a workaround...

In my current tests structure I have
before() - to create precondition with log in and other suff
beforeEach() - located in index.js to load routes
beforeEach() - in describe section of spec file to make another precondition for this particular test
it() - the test itself

So apparently, two beforeEach() methods creating some conflict resulting in the random error I have mentioned above. When I have removed second beforeEach() of spec file and put that logic in every it(), the problem was resolved. Now the tests not crashing because of the routes. It s added the code duplication, because I have copy/pasted precondition from beforeEach() into every test, but that worked and this is all I need.
